components/match2/wikis/callofduty/match_group_input_custom.lua

```diff
@@ -0,0 +1,500 @@
+end
+
+-- Get the winner when resulttype=default
+function CustomMatchGroupInput.getDefaultWinner(table)
+	for index, scoreInfo in pairs(table) do
+		if scoreInfo.status == 'W' then
+			return index
+		end
+	end
+	return -1
+end
+
+--
+-- match related functions
+--
+function matchFunctions.getBestOf(match)
+	match.bestof = Logic.emptyOr(match.bestof, Variables.varDefault('bestof', _DEFAULT_BESTOF))
+	Variables.varDefine('bestof', match.bestof)
+	return match
+end
+
+-- Calculate the match scores based on the map results (counting map wins)
+-- Only update a teams result if it's
+-- 1) Not manually added
+-- 2) At least one map has a winner
+function matchFunctions.getScoreFromMapWinners(match)
+	local opponentNumber = 0
+	for index = 1, _MAX_NUM_OPPONENTS do
+		if String.isEmpty(match['opponent' .. index]) then
+			break
+		end
+		opponentNumber = index
+	end
+	local newScores = {}
+	local foundScores = false
+
+	for i = 1, _MAX_NUM_MAPS do
+		if match['map'..i] then
+			local winner = tonumber(match['map'..i].winner)
+			foundScores = true
+			if winner and winner > 0 and winner <= opponentNumber then
+				newScores[winner] = (newScores[winner] or 0) + 1
+			end
+		else
+			break
+		end
+	end
+
+	for index = 1, opponentNumber do
+		if not match['opponent' .. index].score and foundScores then
+			match['opponent' .. index].score = newScores[index] or 0
+		end
+	end
+
+	return match
+end
+
+function matchFunctions.readDate(matchArgs)
+	if matchArgs.date then
+		local dateProps = MatchGroupInput.readDate(matchArgs.date)
+		dateProps.hasDate = true
+		return dateProps
+	else
+		return {
+			date = mw.getContentLanguage():formatDate('c', _EPOCH_TIME),
+			dateexact = false,
+		}
+	end
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'team'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+
+	match.links = {}
+	local links = match.links
+	if match.reddit then links.reddit = 'https://redd.it/' .. match.reddit end
+	if match.cdl then links.cdl = 'https://callofdutyleague.com/en-us/match/' .. match.cdl end
+
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	match.extradata = {
+		mvp = matchFunctions.getMVP(match),
+	}
+	return match
+end
+
+-- Parse MVP input
+function matchFunctions.getMVP(match)
+	if not match.mvp then return {} end
+	local mvppoints = match.mvppoints or 1
+
+	-- Split the input
+	local players = mw.text.split(match.mvp, ',')
+
+	-- Trim the input
+	for index, player in pairs(players) do
+		players[index] = mw.text.trim(player)
+	end
+
+	return {players=players, points=mvppoints}
+end
```

Comment:
suggest to take a look at how LOL does mvp stuff
it has some additional functionality

Reply:
I've decided to replaced this with **mvp = MatchGroupInput.readMvp(match),** from LoL 

but It seems to return error when MVP input is being attempt on tournament pages saying **bad argument #1 to 'pairs' (table expected, got nil).** on Line 413. I'm not sure if it needs to rewrite those parse player section.

Reply:
please provide an example page with that error (userspace using /dev???)

Reply:
ok, this one - https://liquipedia.net/callofduty/User:Hesketh2/Test

Reply:
https://liquipedia.net/callofduty/index.php?title=Module%3AMatchGroup%2FInput%2FCustom%2Fdev&type=revision&diff=108026&oldid=108021

but you also have to adjust match/legacy and matchsummary modules